src/services/formatters/YScopeFormatter/index.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    BACKSLASH_REGEX,
+    FIELD_PLACEHOLDER_REGEX,
+    Formatter,
+    FormatterOptionsType,
+    YScopeFieldFormatter,
+    YScopeFieldPlaceholder,
+} from "../../../typings/formatters";
+import {LogEvent} from "../../../typings/logs";
+import {
+    getFormattedField,
+    splitFieldPlaceholder,
+    YSCOPE_FORMATTERS_MAP,
+} from "./utils";
+
+
+/**
+ * A formatter that uses a YScope format string to format log events into a string. See
+ * `YScopeFormatterOptionsType` for details about the format string.
+ */
+class YScopeFormatter implements Formatter {
```

Comment:
nit picking - any better names than YscopeFormatter?

Reply:
Discussed offline - it seems we can't find a better name than `YscopeFormatter` at the moment. Let's stick with `YscopeFormatter` then.